contracts/strategies/morpho/MetaMorphoStrategy.sol

```diff
@@ -0,0 +1,159 @@
+
+    rewards = new uint256[](_rewardsLen);
+    bytes[] memory claimDatas = abi.decode(data, (bytes[]));
+    address reward;
+    address rewardDistributor;
+    uint256 claimable; 
+    bytes32[] memory proof;
+
+    for (uint256 i = 0; i < _rewardsLen; i++) {
+      if (claimDatas[i].length == 0) {
+        continue;
+      }
+      (reward, rewardDistributor, claimable, proof) = abi.decode(claimDatas[i], (address, address, uint256, bytes32[]));
+      rewards[i] = _claimReward(IUniversalRewardsDistributor(rewardDistributor), reward, claimable, proof);
+    }
+  }
+
+  /// @notice transfer rewards to IdleCDO, used if someone claims on behalf of the strategy
+  /// or to transfer MORPHO once it's transferable
+  /// @dev Anyone can call this function
+  function transferRewards() public {
+    address[] memory _rewardTokens = rewardTokens;
+    address cdo = address(idleCDO);
+    uint256 _rewardsLen = _rewardTokens.length;
+    IERC20Detailed _reward;
+    for (uint256 i = 0; i < _rewardsLen; i++) {
+      _reward = IERC20Detailed(_rewardTokens[i]);
+      uint256 bal = _reward.balanceOf(address(this));
+      if (bal > 0 && (address(_reward) != MORPHO || morphoTransferable)) {
+        _reward.safeTransfer(cdo, bal);
+      }
+    }
+  }
+
+  /// onlyOwner methods
+
+  /// @dev set to array(0) to skip `redeemRewards`
+  /// @param _rewardTokens array of reward tokens
+  function setRewardTokens(address[] memory _rewardTokens) external onlyOwner {
+    rewardTokens = _rewardTokens;
+  }
+
+  /// @dev used to upgrade mmSnippets contract
+  /// @param _mmSnippets new snippets contract address
+  function setMMSnippets(address _mmSnippets) external onlyOwner {
+    mmSnippets = _mmSnippets;
+  }
+
+  /// @dev set morpho transferability
+  /// @param _isTransferable true if MORPHO is transferable
+  function setMorphoTransferable(bool _isTransferable) external onlyOwner {
+    morphoTransferable = _isTransferable;
+  }
+
+  /// Internal methods
+
+  /// @notice claim and transfer reward to idleCDO by verifying a merkle root
+  /// @param distributor contract for distributing rewards
+  /// @param reward reward address to claim
+  /// @param claimable amount of reward to claim
+  /// @param proof merkle proof
+  function _claimReward(
+    IUniversalRewardsDistributor distributor,
+    address reward,
+    uint256 claimable,
+    bytes32[] memory proof
+  ) internal returns (uint256 claimed) {
+    uint256 balBefore = IERC20Detailed(reward).balanceOf(address(this));
+    distributor.claim(address(this), reward, claimable, proof);
+    claimed = IERC20Detailed(reward).balanceOf(address(this)) - balBefore;
+    // MORPHO is not transferable atm
+    if (reward != MORPHO || morphoTransferable) {
+      IERC20Detailed(reward).safeTransfer(address(idleCDO), claimed);
+    }
+  }
+
+  /// View methods
+
+  /// @notice return the reward tokens
+  function getRewardTokens() external override view returns (address[] memory) {
+    return rewardTokens;
```

Comment:
it feels redundant to have this getter and `rewardTokens` public, but I understand both usecases. Ideally, you can adopt our way of handling this: [extsloads](https://github.com/morpho-org/morpho-blue/blob/731e3f7ed97cf15f8fe00b86e4be5365eb3802ac/src/Morpho.sol#L543-L556) (inspired from uni v4)

Reply:
We need this to be compatible with the other strategies (and the interface that we need to implement) where sometimes there a different logic is needed